src/scenes/level_scene.py

```diff
@@ -2198,6 +2212,19 @@ def button_down(self, button: int, position: Position) -> None:
                                     isinstance(entity, Character),
                                 )
                             return
+                        
+    def key_down(self, keyname):
+        """
+        Handle the triggering of a key down event.
+
+        Keyword arguments:
+        keyname -- an integer value representing which key button is down
+        """
+        if keyname == pygame.K_ESCAPE:
+            if (self.menu_manager.active_menu is not None
+                and self.menu_manager.active_menu.title != "Select an action"
```

Comment:
Good idea to block such menus, but I don't think it's ever a good idea to rely on strings displayed on the screen.
Because it means that if one day someone wants to make a subtle change to this text, the application would be broken. Also, what about the multilingual support you are going to introduce? It may not be this text on the screen, but its translation.

I suggest two alternative solutions:
* `Infobox` has attribute `has_close_button`, we could assume that if a modal doesn't have any close button, then it means it cannot be close, because it's a blocking menu (waiting for user action). That would be the generic approach.
* `Infobox` also has attribute `identifier`, this identifier could be used to refer to the specific "select an action" modal instead of relying on something that may change. So the solution would be to give a meaningful id at the creation of the modal, and at this place check if the `active_menu` matches the given id. That would be a more specific approach.

Btw, if you need any help regarding how the UI components are working: [https://pygame-popup-manager.readthedocs.io/en/latest/api/components/info_box.html](https://pygame-popup-manager.readthedocs.io/en/latest/api/components/info_box.html)

Reply:
Conditional judgement in Line2225 copied the original judgement in Line2132(in function right_click). I had no intention of resolving the above TODO at the time, so I copied it directly. Maybe we can handle it later. 

Another fact is that when multilingual support added, it will not be an issue cuz the code is going to be like this:
```python
if self.menu_manager.active_menu.title == STR_SELECT_AN_ACTION:
```

Reply:
My bad, I didn't notice it was already this way before!
This TODO has to be from a time where  `InfoBox` didn't have any id haha.

You are definitely right then, it should be handled in another PR since it is not introduced by this change.
I'm going to take care of it.